### ISSUE TITLE
Failed to generate CHANGELOG.md

### ISSUE THREAD

**Issue by kieronlanning** [easybuild-org/EasyBuild.ChangelogGen#3]
When running `dotnet changelog-gen`, I get the following:

```bash
Error: Invalid commit message format.

Expected a commit message with the following format: '<type>[optional scope]: <description>'.

Where <type> is one of the following:

- feat: A new feature
- fix: A bug fix
- ci: Changes to CI/CD configuration
- chore: Changes to the build process or auxiliary tools and libraries such as documentation generation
- docs: Documentation changes
- test: Adding missing tests or correcting existing tests
- style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- refactor: A code change that neither fixes a bug nor adds a feature
```

When I run `git log` I get the following:

```bash
commit f8918cf8d14db87a48282134670473f2f8d937ce (HEAD -> main, origin/main)
Author: Kieron Lanning <kieronlanning@users.noreply.github.com>
Date:   Fri Nov 29 12:04:21 2024 +0000

    build: initial project commit
```

Note the only commit is `build: initial project commit`.

I'm linting/ validating the commands based on `conventional-changelog/commitlint` and their default [list](https://github.com/conventional-changelog/commitlint/?tab=readme-ov-file#what-is-commitlint) which in turn uses the popular [Angular](https://github.com/conventional-changelog/commitlint/tree/master/@commitlint/config-conventional#type-enum) convention.

**Comment by MangelMaxime** [comment]
For some reason, when implementing the default config I didn't implement all the commit type proposed by Angular convention.

It is missing `build`, `perf`.

Trying to remember the reason, I think I didn't implement them because I thought:

- `build` is covered by `chore` when looking at the description
- `perf`, should be covered by `feat` or `feat(perf)` because if you make a `perf` commit it will not be included in the changelog as only `feat`, `fix` and `breaking change` trigger a version bump

Adding `build` to the default configuration is not too difficult. However I have more problem with `perf` because like mentioned above if we make a `perf` commit it will not be included in the changelog and so will not trigger a new release.

But I believe, if we make a `perf` improvement you want it to trigger a new release, don't we?

A way to workaround that would be to re-introduce a feature that I trimmed down, which is to allow the user to configure if a commit type should trigger a version bump or not. https://github.com/easybuild-org/EasyBuild.ChangelogGen/issues/2

---

Currently, you can avoid the error by using `--skip-invalid-commit` which will just ignore the invalid commits type when generating the changelog. It was mostly introduce for people migrating a old repo to the Conventional Commit format or also to ignore commit generating by Github when merging branch, etc.

A cleaner solution, is to implement a config file allowing your own type of type/scope. The configuration format is defined [here](https://github.com/easybuild-org/EasyBuild.CommitParser#configuration) and you can use the `--config` options to pass your configuration to the changelog generator.

**Comment by kieronlanning** [comment]
Thanks for getting back to me.

I can see your point about `feat(perf)`, but at the same time it would preclude anything such as `perf(engine): xxxxx` where it's not a new or changed feature, per-se, but a performance improvement only. One what might span more than just one part of the application. 

I the Angular list appears to be the _industry standard_.

**Comment by MangelMaxime** [comment]
> I can see your point about `feat(perf)`, but at the same time it would preclude anything such as `perf(engine): xxxxx` where it's not a new or changed feature, per-se, but a performance improvement only. One what might span more than just one part of the application.
> 
> I the Angular list appears to be the _industry standard_.

I understand, but then what is your expectation about `perf` type commit?

Do you expect it to be included in the changelog? to generate a version bump?

**Comment by MangelMaxime** [comment]
Looking at angular convention it seems like the situation at the moment is:

Supported types:

- chore
- ci
- docs
- feat
- fix
- refactor
- style
- test

Not supported:

- build
- perf
- revert

I am ok with adding `perf` and I think it can be treated like `feat` it will do a minor bump and put commits under their own section `🏎️ Performance Improvements`.

This could be revisited in the future, if I introduce a system like described in #2 and a concept of preset. Like that people could chose to use Angular preset or a simpler preset, etc.

For now, I will go with the easiest path of adding it to the default configuration.

**Comment by MangelMaxime** [comment]
I released version 3.0.0:

- `build`, `perf` and `revert` commit type should now be supported
- a new `⚡ Performance Improvements` sections has been added to the changelog and if you have a `perf` commit type it will bump the minor version

EasyBuild.CommitParser and EasyBuild.CommitLinter have also been updated to reflect this change.

**Comment by kieronlanning** [comment]
Amazing, thanks @MangelMaxime! \o/
